src/dir.rs

```diff
@@ -67,6 +67,40 @@ impl Dir {
     pub fn iter(&mut self) -> Iter {
         Iter(self)
     }
+
+
+    /// Set the position of the directory stream, see `seekdir(3)`.
+    #[cfg(target_os = "linux")]
+    pub fn seek(&mut self, loc: SeekLoc) {
+        unsafe { libc::seekdir(self.0.as_ptr(), loc.0 as libc::c_long) }
```

Comment:
On 32-bit OSes `long` is usually 32-bit. According  to this [stackoverflow post](https://stackoverflow.com/questions/9073667/where-to-find-the-complete-definition-of-off-t-type) `off_t` should also be 32-bit on 32-bit OSes. 
But I remember I had to add some casts yesterday to satisfy the compiler for arm-linux-gnueabi, so I'm quite confused here.

Btw. is there any good reference on how wide the integer types are on different platforms (say, for glibc)?
Probably using `long` would be fine here, because the value can only be used through `seekdir`

Reply:
> Then that would be a bug in either Linux or glibc. telldir is not allowed to fail.

@asomers I did a small revision to my previous post while you were typing. According to my analysis, this would just result in UB.

Reply:
I switched to `off64_t` in `SeekLoc`, so that the only place where the truncation can occur is the call to `seekdir` (but this is the case when the truncation would occur anyway in C code, either in `seekdir` or in `telldir`)

Reply:
That doesn't solve the real problem, which is that you're using `d_off` when you should be using `seekdir`.

Reply:
What's wrong with exposing both ways to get the seek location, if the C library also exposes them both?

Reply:
Related: https://sourceware.org/ml/libc-help/2019-08/msg00007.html

Reply:
`d_off` is non-portable and has no advantages over using `telldir`.  Since an `Iterator` already has a reference to its parent `Dir`, what's wrong with using `telldir`?

Reply:
It has only a performance advantage. I suggest we remove `d_off` for now and potentially discuss it's inclusion in a subsequent pull request.

Reply:
I doubt there's any measurable performance advantage.  glibc's telldir probably just checks `d_off`.